src/main/scala/org/ergoplatform/mining/ErgoMiner.scala

```diff
@@ -88,16 +88,17 @@ class ErgoMiner(ergoSettings: ErgoSettings,
   private var secretKeyOpt: Option[DLogProverInput] = inSecretKeyOpt
 
   // "miningPubkeyHex" setting in config has preference over wallet's secret key
-  private var publicKeyOpt: Option[ProveDlog] = ergoSettings.miningPubKey.orElse(inSecretKeyOpt.map(_.publicImage))
+  private var publicKeyOpt: Option[ProveDlog] = ergoSettings.miningPubKey
 
   // internal (CPU) miner threads, if internal mining is chosen
   private val miningThreads: mutable.Buffer[ActorRef] = new ArrayBuffer[ActorRef]()
 
   override def preStart(): Unit = {
-    // in external miner mode key from wallet is used if `publicKeyOpt` is not set
-    if ((publicKeyOpt.isEmpty && externalMinerMode) || (secretKeyOpt.isEmpty && !externalMinerMode)) {
+    if (secretKeyOpt.isEmpty && !externalMinerMode) {
+      self ! QueryWallet(secret = true)
+    } else if (publicKeyOpt.isEmpty) { // mining pubKey is needed in both modes
```

Comment:
I removed `externalMinerMode` condition here as PK is needed in both modes, that's why all `ErgoMinerSpec` tests were failing, PK was absent and no candidate was generated ...

Reply:
Good catch!